common/utils/typeconverter.go

```diff
@@ -71,9 +75,74 @@ func SegmenterValueToInterface(value *_segmenters.SegmenterValue) interface{} {
 	}
 }
 
-func InterfaceToSegmenterValue(value interface{}, valueType *_segmenters.SegmenterValueType) *_segmenters.SegmenterValue {
+func GetIntSegmenter(value interface{}, key string, segmenter string) (*int64, error) {
```

Comment:
`GetIntSegmenter`, `GetFloatSegmenter`, `GetBoolSegmenter` adds support for casting string request values to appropriate segmenter type values.

Reply:
Cool. Although now, the implementation looks pretty bulky. Shall we instead use [this library](https://github.com/spf13/cast)? It seems to cater to the corner cases too. If using the library, may be we don't need the utility methods anymore and can be directly called where required.

Reply:
I think the cast  package is definitely useful, but it does catch some corner cases which are not what we want. 

**Examples**
```
cast.ToFloat(true)            // 1
cast.ToFloat(false)          // 0
cast.ToInt(true)               // 1
cast.ToInt(false)              // 0
```
For the above cases, our current implementation raise errors for us, which is what we want, rather than casting to binary values.

Reply:
Fair enough. But:
* Should we really worry about the user registering an int type and sending in a boolean value?
* Can this be considered a feature of XP rather than a bug? 😅 We can argue that `True: 1` and `False: 0` are conventional?

Reply:
>  Should we really worry about the user registering an int type and sending in a boolean value?

I'd say its user problem, just that we won't be providing any informative feedback loop to the user immediately, they'd likely have to debug themselves when they notice treatments are not fetched correctly.

> Can this be considered a feature of XP rather than a bug? 😅 We can argue that True: 1 and False: 0 are conventional?

I guess we could, I see it as a tradeoff between ease of debugging vs. maintenance, and in this case, it does feel more like a user problem than something we should be catching 🤔 

We'll go with `cast` then?

Reply:
Yes, let's reduce our maintainable code considering the problem is an atypical scenario.